nav2_mission_executor/README.md

```diff
@@ -0,0 +1,82 @@
+# Mission Executor
+
+The Mission Executor module is a task server that coordinates other tasks. It can be used to direct the activities multiple robots as well as other, non-robot tasks.
+
+## Overview
+ 
+The Mission Executor task server receives a **MissionPlan** and returns whether the plan was successfully executed, failed, or was canceled.
+
+```C++
+namespace nav2_tasks {
+
+using ExecuteMissionCommand = nav2_msgs::msg::MissionPlan;
+using ExecuteMissionResult = std_msgs::msg::Empty;
+
+using ExecuteMissionTaskClient = TaskClient<ExecuteMissionCommand, ExecuteMissionResult>;
+using ExecuteMissionTaskServer = TaskServer<ExecuteMissionCommand, ExecuteMissionResult>;
+
+} //  namespace nav2_tasks
+```
+
+NOTE: An Empty message is used in the definition of the task client and server if there is no other data required.
+
+The [mission plan message](https://github.com/ros-planning/navigation2/blob/master/nav2_msgs/msg/MissionPlan.msg) consists of a header and a mission plan string:
+
+```
+std_msgs/Header header
+string mission_plan
+```
+
+The mission plan itself is an XML string that defines the behavior tree. For example, the following XML string specifies a sequence of three NavigateToPose actions:
+
+```XML
+<root main_tree_to_execute="MainTree">
+  <BehaviorTree ID="MainTree">
+    <SequenceStar name="root">
+      <NavigateToPose position="10;11;0.0" orientation="0.7071;0;0.7071;0"/>
+      <NavigateToPose position="20;21;0.0" orientation="0.7071;0;0.7071;0"/>
+      <NavigateToPose position="30;31;0.0" orientation="0.7071;0;0.7071;0"/>
+    </SequenceStar>
+  </BehaviorTree>
+</root>
+
+```
+
+The mission plan module uses the [Behavior-Tree.CPP library](https://github.com/BehaviorTree/BehaviorTree.CPP) to dynamically create a behavior tree from the input XML description. It then generates and executes the tree and returns the status code (SUCESSFUL, FAILED, CANCELED) to the task client.
+
+## Incorporating Recovery Behaviors
+
+The behavior tree can incorporate recovery actions by responding to conditions in the tree with other actions; a failure of one action could result in the execution of a recovery action, for example. 
+
+## Creating Behavior Tree Nodes
+
+To create a node that can be included in a behavior tree, there must first be an task server/client defined for it. For example, the NavigateToPose task is defined as follows:
```

Comment:
Potentially unrelated to this specific file -- but can the NavigateToPose be called easily without defining the whole mission task? For instance, in ROS1 we had "go to" button in RVIZ that simply sends a pose to a non-action interface in MoveBase. This is a highly used interface, especially when testing on a new robot.

Reply:
Also useful when the impact of what was completed / seen / done at that waypoint effects what the next action is. Most intelligent robotics applications I don't think you can just treat as a point and shoot to a set of predetermined points.

Reply:
@mikeferguson Yes, any of the tasks can be called directly. One could simply call NavigateToPose with the desireed target pose. The mission planner is a layer above that to coordinate various tasks, but is not required. We're looking at coordinating multiple robots and tasks and what that would look like. 

@SteveMacenski Yes, I agree. I'm thinking that Behavior Trees would be useful to capture the more complex behaviors. BTs could be used at the mission planning/execution level to coordinate various high-level tasks and can also be used to implement specific tasks. In either case, I expect that it can be very dynamic and respond to outputs.